stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -352,19 +317,16 @@ function mul!(C::AbstractMatrix, Da::Diagonal, Db::Diagonal, alpha::Number, beta
     return C
 end
 
-(/)(Da::Diagonal, Db::Diagonal) = Diagonal(Da.diag ./ Db.diag)
-
-ldiv!(x::AbstractArray, A::Diagonal, b::AbstractArray) = (x .= A.diag .\ b)
+(/)(A::AbstractVecOrMat, D::Diagonal) =
+    _rdiv!(similar(A, promote_op(/, eltype(A), eltype(D)), size(A)), A, D)
+(/)(Da::Diagonal, Db::Diagonal) = Diagonal(vec(permutedims(Da.diag) / Db))
```

Comment:
What's the performance of this both for small and large cases? Once we're at it, maybe we have another `_rdiv!` method for the two-Diagonal case? It should be quite a couple of lines shorter than the general one.

Reply:
At least `permutedims` should do harm to small case.
I also wonder whether the compile know the first dimension is one sized. If not there should be more performance regression.

Implement this in the deepest fallback call is a good idea. As `rdiv!` for this case should also be included automatically. Similar for 3arg `ldiv!`

Reply:
Found that `similar(::Diagnal, ::Type, size(A))` returns a sparse matrix.
Is there any reason we use this rather than `similar(::Diagnal, ::Type)`?